src/script/storage/StorageService.ts

```diff
@@ -115,24 +116,13 @@ export class StorageService {
     }
   }
 
-  private async moveDexieToMemory(): Promise<void> {
-    const objectDb: MemoryStore = {};
-
-    for (const table of this.db.tables as Dexie.Table<Record<string, any>, string>[]) {
-      const keys = await table.toCollection().keys();
-      objectDb[table.name] = {};
-      for (const key of keys.map(key => key.toString())) {
-        objectDb[table.name][key] = await table.get(key);
-      }
-    }
-
-    await this.engine.initWithObject(this.dbName, objectDb);
-    this.objectDb = objectDb;
-
-    await this.db.delete();
-    await this.db.close();
-
-    this.db = undefined;
+  /** @deprecated Use `provideTemporaryAndNonPersistentEngine` instead */
+  static initEncryptedDatabase(encryptionKey: string): SQLeetEngine {
+    return new SQLeetEngine(
```

Comment:
Would it make sense to move the logic for getting the key in here?

Reply:
Getting the key has been already outsourced to `saveRandomEncryptionKey` function. I think it's okay to use this one here for the moment.